### PR TITLE
add argument reading for loading ROM file

### PR DESCRIPTION
**Pull request by tiopex**
- Use SDL_HWSURFACE and SDL_DOUBLEBUF overall performance increased
-add argument reading for loading ROM file.
-fix alsa reinit

### CODE REVIEW

**Comment by tiopex** [comment]
build with static musl is faster than uclibc

**Comment by Apaczer** [comment]
How about static uClibc? This is the one bittrider have used, albeit custom build with removed dependecies and gcc7 (see release page).

**Comment by tiopex** [comment]
on this SDK it is fast as bitrider release

**Comment by tiopex** [comment]
I reverted flags, as the originals are the best on SDK with gcc7, which have best performance

File: Makefile [hunk 1]
```diff
@@ -71,13 +71,13 @@ wiz_common: SDK = $(CROSS_ROOT)/usr
 wiz_common: do
 
 # -- Fast version
-wizf: VNAME = $(NFAST)
+wizf: VNAME = $(NFAST)
```
Comment (Apaczer) [inline Makefile]: Watch out for trailing spaces! They brake Makefile instructions.

File: src/miyoo_sdk.c [hunk 2]
```diff
@@ -373,10 +373,10 @@ void gp_initGraphics(unsigned short bpp, int flip, int applyMmuHack)
 
 	if (screen == NULL)
 {
-	screen = SDL_SetVideoMode(320, 240, bpp, SDL_SWSURFACE|SDL_FULLSCREEN);
-	if ( screen == NULL ) {
-		fprintf(stderr, "Unable to set 320x240 video: %s\n", SDL_GetError());
-		exit(1);
+	screen = SDL_SetVideoMode(320, 240, bpp, SDL_HWSURFACE|SDL_FULLSCREEN|SDL_DOUBLEBUF);
```
Comment (Apaczer) [inline src/miyoo_sdk.c]: tabs pls

File: src/miyoo_sdk.c [hunk 3]
```diff
@@ -413,7 +413,6 @@ void gp_initGraphics(unsigned short bpp, int flip, int applyMmuHack)
 
 void gp_setFramebuffer(int flip, int sync)
 {
-	CurrentFrameBuffer=flip;
```
Comment (Apaczer) [inline src/miyoo_sdk.c]: why not just leave that assigment?

**Comment by Apaczer** [comment]
Fine be me, if it's on the same lvl as previous (closed) release, which is, in fact the same 👍.

File: src/miyoo_sdk.c (hunk 3 above)
Comment (tiopex) [reply src/miyoo_sdk.c]: not needed

**Comment by tiopex** [comment]
I checked with `-fopt-info-optimized` what are the differences in optimizations between gcc 9.x and gcc 7.x and found that 9.x performs also inlining.  After adding flag `-fno-inline` performance increased a lot and it is even better than previous release

used miyoocfw/toolchain-static-uclibc:latest SDK

**Comment by Apaczer** [comment]
> used miyoocfw/toolchain-static-uclibc:latest SDK

Does static outperforms dynamic linking? Thanks for `-fopt-info` notion (by default in -optimized mode?), good tool to bisect optimization and crossout any redundant flags.

**Comment by tiopex** [comment]
Didn't test it, as I'm focus only on static SDKs, to avoid create separate ipk for uclibc/musl

**Comment by Apaczer** [comment]
well with PGO, the legacy build has been put to a shame. May vary from running scenario, but tested with one game back and forth, with almost 8fps boost

File: src/main.cpp [hunk 4]
```diff
@@ -1362,9 +1372,26 @@ int main(int argc, char *argv[])
 
 	if (!S9xGraphicsInit ())
          erk();
+	// read command line options
+	for (i = 1; i < argc; i++) {
+		if (!strcmp(argv[i], "-rom")) {
```
Comment (Apaczer) [inline src/main.cpp]: remove this option, load ROM by default and allow only for 1 argument or output stderr

**Comment by Apaczer** [comment]
There are still audio/values distortions, for e.g. in MarioKart with values:
```
bittrider - 367 ; 1468
ours - 441; 1764
```

**Comment by tiopex** [comment]
UAre you sure that bitrider has 367 not 376?
The sample is calculated by rate / fps +1

So for PAL and rate 22500
22500 / 50 = 375 + 1

And for ntsc
22500 / 60 = 450 + 1

These your values are really weird. Which version you are testing, EU, or USA?

**Comment by Apaczer** [comment]
367 (rechecked) and USA. Inline with what u saying but 22050 that is and for 60 frame rate

**Comment by tiopex** [comment]
yes, my fault, I checked also, and now I have an identical values and correct sound

our: 
```

"SUPER MARIO KART" [checksum ok] HiROM, 4Mbits, Type: ROM+RAM+BAT+DSP1, Mode: 21, TV: NTSC, S-RAM: 2Kb, ROMId:  Company:
Actual Audio Freq 22050 Period 367 bufsize 1468 period_time 16643 frame_limit 60
```

bitrider
```
rate: 22050 bps
period sz: 367 frames, 1468 bytes
period time: 16643
```

**Comment by Apaczer** [comment]
Did a quick run in QEMU and LGTM! (didn't output snd while running as child from gmenu2x but it might be qemu specific). Will test on real hw at evening

**Comment by Apaczer** [comment]
I believe ScanSaveStates() doesn't properly read currentRomFilename while using argument, cuz all savestates gets blank names.

**Comment by tiopex** [comment]
I thought there would be another case that I hadn't checked

File: src/main.cpp [hunk 5]
```diff
@@ -1197,6 +1207,9 @@ int main(int argc, char *argv[])
 	int action=0;
 	int romloaded=0;
 	char text[256];
+	char file[MAX_PATH+1] = "";
```
Comment (Apaczer) [inline src/main.cpp:1210]: just curious why increase MAX_PATH size ? (PATH_MAX)

**Comment by Apaczer** [comment]
Only minor stuff left TODO, beside some upgrades. Thanks @tiopex for helping in opensourcing this project.

File: src/main.cpp (hunk 5 above)
Comment (tiopex) [reply src/main.cpp:1210]: I reused existed code, in all other places there are max_path +1

**Pull request merged**